api/types.go

```diff
@@ -361,9 +361,9 @@ type CopyRequest struct {
 // PullRequest is the request passed to [Client.Pull].
 type PullRequest struct {
 	Model    string `json:"model"`
-	Insecure bool   `json:"insecure,omitempty"`
-	Username string `json:"username"`
-	Password string `json:"password"`
+	Insecure bool   `json:"insecure,omitempty"` // Deprecated: ignored
```

Comment:
What will happen on pulling from an http source now?

Reply:
It just pulls, like curl would, or wget, or literally anything else.

Reply:
I've added `https+insecure` for pulling/pushing to remotes that have certs but not in a CA and users are fine skipping verification (e.g. https://localhost:6000).